Sources/NIOHTTP1/HTTPEncoder.swift

```diff
@@ -95,16 +98,21 @@ private func isChunkedPart(_ headers: HTTPHeaders) -> Bool {
 ///
 /// Note that for HTTP/1.0 if there is no Content-Length then the response should be followed
 /// by connection close. We require that the user send that connection close: we don't do it.
-private func sanitizeTransportHeaders(hasBody: HTTPMethod.HasBody, headers: inout HTTPHeaders, version: HTTPVersion) {
+private func sanitizeTransportHeaders(hasBody: HTTPMethod.HasBody, headers: inout HTTPHeaders, version: HTTPVersion) -> BodyFraming {
     switch hasBody {
     case .no:
         headers.remove(name: "content-length")
         headers.remove(name: "transfer-encoding")
-    case .yes where headers["content-length"].count == 0 && version.major == 1 && version.minor >= 1:
+        return .neither
+    case .yes where version.major == 1 && version.minor >= 1:
+        if headers.contains(name: "content-length") {
+            return .contentLength
+        }
         headers.replaceOrAdd(name: "transfer-encoding", value: "chunked")
+        return .chunked
     case .yes, .unlikely:
         /* leave alone */
-        ()
+        return headers["transfer-encoding"].first == "chunked" ? .chunked : .neither
```

Comment:
This is wrong. For HTTP/1.0, there is no chunked transfer encoding. This should look for content-length, not chunked TE. I appreciate it won't affect the outcome of the code, but we should aim to get this right in case we want to use this elsewhere.

Reply:
Yup, chunked + HTTP/1.0 is not spec-valid.